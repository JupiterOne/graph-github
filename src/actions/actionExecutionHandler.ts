import {
  EntityIngestError,
  partialIngestActionHandler,
} from './partialIngestActionHandler';
import {
  Entity,
  IntegrationError,
  IntegrationExecutionContext,
  IntegrationInstanceConfig,
} from '@jupiterone/integration-sdk-core';
import { IntegrationConfig } from '../config';
import { getOrCreateRestClient } from '../client/RESTClient/client';
import { getOrCreateGraphqlClient } from '../client/GraphQLClient';

// TODO: extend IntegrationAction from jupiter-types VDubber 5/2022
interface IntegrationPartialIngestAction {
  name: string;
  parameters: Record<string, any>;
}

export interface IntegrationActionExecutionContext<
  TConfig extends IntegrationInstanceConfig,
> extends IntegrationExecutionContext<TConfig> {
  event: {
    accountId: string;
    integrationInstanceId: string;
    action: IntegrationPartialIngestAction;
    timestamp: number;
  };
}

export default async function actionExecutionHandler(
  context: IntegrationActionExecutionContext<IntegrationConfig>,
): Promise<{ entities: Entity[]; errors: EntityIngestError[] }> {
  const { logger, instance, event } = context;
  const { action } = event;

  logger.debug({ integrationAction: action }, 'Handling integration action');

  const restClient = getOrCreateRestClient(instance.config, logger);
  const graphqlClient = getOrCreateGraphqlClient(instance.config, logger);

  await restClient.verifyAuthentication();

  switch (action.name) {
    case 'PARTIAL_INGEST': {
      if (!action.parameters) {
        throw new Error('parameters are required for action PARTIAL_INGEST');
      }

      return await partialIngestActionHandler(
        graphqlClient,
        action.parameters.entities,
        instance.config,
        logger,
      );
    }
    default: {
      throw new IntegrationError({
        code: 'UNKNOWN_ACTION_NAME',
        message: 'The provided action.name is unknown.',
      });
    }
  }
}
