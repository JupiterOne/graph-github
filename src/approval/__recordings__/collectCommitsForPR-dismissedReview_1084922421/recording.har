{
  "log": {
    "_recordingName": "collectCommitsForPR.dismissedReview",
    "creator": {
      "comment": "persister:SafeHeadersFSPersister",
      "name": "Polly.JS",
      "version": "2.6.2"
    },
    "entries": [
      {
        "_id": "6246ade88b1f3836adedaeafadb21275",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.machine-man-preview+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiter-integration-github octokit-rest.js/17.6.0 octokit-core.js/2.5.4 Node.js/10.19.0 (Linux 4.4; x64)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "Bearer [REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "0"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 813,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "queryString": [],
          "url": "https://api.github.com/app/installations/953957/access_tokens"
        },
        "response": {
          "bodySize": 1183,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 1183,
            "text": "{\"id\":953957,\"account\":{\"login\":\"github-app-test\",\"id\":50627219,\"node_id\":\"MDEyOk9yZ2FuaXphdGlvbjUwNjI3MjE5\",\"avatar_url\":\"https://avatars1.githubusercontent.com/u/50627219?v=4\",\"gravatar_id\":\"\",\"url\":\"https://api.github.com/users/github-app-test\",\"html_url\":\"https://github.com/github-app-test\",\"followers_url\":\"https://api.github.com/users/github-app-test/followers\",\"following_url\":\"https://api.github.com/users/github-app-test/following{/other_user}\",\"gists_url\":\"https://api.github.com/users/github-app-test/gists{/gist_id}\",\"starred_url\":\"https://api.github.com/users/github-app-test/starred{/owner}{/repo}\",\"subscriptions_url\":\"https://api.github.com/users/github-app-test/subscriptions\",\"organizations_url\":\"https://api.github.com/users/github-app-test/orgs\",\"repos_url\":\"https://api.github.com/users/github-app-test/repos\",\"events_url\":\"https://api.github.com/users/github-app-test/events{/privacy}\",\"received_events_url\":\"https://api.github.com/users/github-app-test/received_events\",\"type\":\"Organization\",\"site_admin\":false},\"repository_selection\":\"all\",\"access_tokens_url\":\"https://api.github.com/app/installations/953957/access_tokens\",\"repositories_url\":\"https://api.github.com/installation/repositories\",\"html_url\":\"https://github.com/organizations/github-app-test/settings/installations/953957\",\"app_id\":19640,\"target_id\":50627219,\"target_type\":\"Organization\",\"permissions\":{\"members\":\"read\",\"metadata\":\"read\",\"pull_requests\":\"read\",\"statuses\":\"read\",\"vulnerability_alerts\":\"read\"},\"events\":[],\"created_at\":\"2019-05-14T14:07:05.000Z\",\"updated_at\":\"2019-05-29T17:52:38.000Z\",\"single_file_name\":null,\"token\":\"[REDACTED]\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 20 Sep 2019 17:36:01 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "status",
              "value": "200 OK"
            },
            {
              "name": "cache-control",
              "value": "public, max-age=60, s-maxage=60"
            },
            {
              "name": "vary",
              "value": "Accept, Accept-Encoding"
            },
            {
              "name": "etag",
              "value": "W/\"37642adf0258a5486c4019eb6b772223\""
            },
            {
              "name": "x-github-media-type",
              "value": "github.machine-man-preview; format=json"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "x-github-request-id",
              "value": "A26D:25FD:60F32A:A9077E:5D850E00"
            }
          ],
          "headersSize": 968,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2019-09-20T17:36:00.728Z",
        "time": 377,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 377
        }
      },
      {
        "_id": "30118b21fd591e65d2a4a6ef18b856b6",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.v3+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiter-integration-github octokit-rest.js/17.6.0 octokit-core.js/2.5.4 Node.js/10.19.0 (Linux 4.4; x64)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "Bearer [REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 363,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "per_page",
              "value": "250"
            }
          ],
          "url": "https://api.github.com/repos/github-app-test/power-demo/pulls/4/commits?per_page=250"
        },
        "response": {
          "bodySize": 3856,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 3856,
            "text": "[\"1f8b0800000000000003ed595957eac816fe2b2c5ed593a432bbfadcdb619e1204122036bdceaacc0919301382cbffde154044da5650fbde1779022a7ba85dfbdb7b7f953f1eca89\",\"03cbd765dce2398003c0681c4fb1e8176021200c4db32cdad275c07396c5f0045bbe2c879161fe720d2424d6546600f8549bf8b828dba4e8b5c9be3c5ff7bd4ad097ed95586bafa4a04df5e5012936555af40c47f41a7335e804b7de385081b8eacb0efa8ed664fd2752ae4741e0a6e5eb8732cc52278a8b6f210c4c64ad9d40a89726aeefbb3048d0b366005d1f2db8c5c2efa9632e776b3f96c5b201d3420ce0047f85d357809709ee1a806bc0dc961f9f2ca5e6a189a69bb632ed407518c5e6c25ffd6ebba993693f9077ef2b0ecc248176615a308c92bf8a5d3d29a5516958176a621dc9a7b189569f220f2d5eb7348ae678dca2599a02384be8aca193baa6a315dd02106804c321c12c2eb6eba4e922b9c630b8707f3cfb852147a304dbfe7105178babd44c526c112dcdf8ca3083a858c20ad30976b24914a6cfdbdc1e69829d9162858819a6bff4280b5132e097e5dc8c5dcbd561ea466111bbed6f1365611a67e66539366152ac9473e8a3dcbc2c27ae1dc2348b8b73b82a3e957ab32d956e9a37a551bb2909b232ac6ffe9f85b3709954f48a2054aa8230a80c3a3a9b906e7558e9b28e47f573326e2f05c168b5db42b7860b792b73c7e32e2f9b4d0f66fa14aef9591837d3c4a5fa42951e27f8a43e9f13d9d88a80d26a7a8691a4da609d6b9a279b44778877bb7050a3ee62e04e710adea95350473edcc4869311aa712fc72de9f63ec66eb0bbf1382393fe7a44c7e305a6db9d55df8973ab4929bc54d54445ea089200d5b9586f4e1db48b3a68b9cb96c52a174bb9a33494fb9a47dd75042e57c85c8ff33b753202f478a5a8382bb41df75ec857dd889f1216102bfadc9d85ade1bc67dcb352d7bfb0c91e198463fa026348c3910c694cb5adcea83b59f16b8ad2eff5ae3e8ee7726549252a16e74e96367b8359a8826e6d5155d37acb31e79032292f5b57145def362c613418f02b49bc70e6f494c303fb36e4474bbf625ad3881c0ffb61a74ffd9c853f2b11d79e859bb3a94bb5574f0c9def02aefc081655a8c8e9d2a9293d0b173046a955b208dc34098a324c54d9004b690ca00d60f03c616a348fd3042001ceb1fa2cdc16a2d2cbf253faedb5aaf39f1241d33c415304c594ae700ac767e136fd519d296dab4be9b7bf1795d7e466e1abe5e3f1d398fc001e9d34f07fbdac040795e78d9ab3b5f501e82747e6ce2d76e76e12db959ca2713c771e3fb2dda2a85851518e5c88168bc647d080a128fa652b1c30e3a9e4eb5e9d103d15976a4ad1ce600e53181fef65f36742eeaa779698b11e8529b25034182cc376faff9bffa4900e3bde69d9f4dcf7da40a12dc10e1c7efbec0e1eb422df2fbac53ba17f690059da4921c7b6dfddd03edef0cb4ef5aa0624f58045a889a368a12d3c161b7793f43c6736120fa8d325299a500a1d098a746c1a6739b49341ee2c43e4c9c3a6b56e94655aa2c7eea26842e739f64212698a621b86ee7ad3ceced384248b1cdd74fbb376b5914092668e12ed3cd1adc803b688dd1ceaab2214b1a99b6e8e02fb017547b2485bba5a147d5a41075f84d94dcd5fd0080ae859d04fcce369ed09964b53bbb250ce22990d2c790ec19262ff019774bf56e07240a2c73f88cb9d814f00f3c0e5b78179f0e059c07c92fb38308f347c06987b559f02e65ecbd701f359e521a43f02ccbda67381b9173c1f987bd1af01e681272f407d1230b7e35452befee389d09c3a5921f59f23174f3dfe0c836f63eefd41e664538f7f3e5eee0342311c600c0d420dd23a0d2c868404d400c50382c639868724c341bde09d2770eb001531cf46bc7ae8aa9386abca634f923b81e4498e887e8bde00f16c3484d46cd09fd4d792e7382a68ff4fb835754d91ff0ab7de2b3ee2d630494bbe1b9a45a6be20d53807348b\",\"85246f418b304ccbd419dcd04d146b9a8214ade32ccb6a1a8d7f3a079f49f5c9263f3dc01736f7b97f7a6eed26dcff23a9a6e357497533c6a7095881c67c3d5d352bbd01d7b2878a3a0bf17a18374ca5da9fc07947631441765510fb5c9a70b7e672903630bf450d191c342a632bd52a9dd468a4c37a8323862305d46ab390e935971732d5cbe64a67d48bc1a2460edaed7010e1b6e0f29dbbe896ab7832c6fab2aaaaed1bfe76ed385dd5e94f72d2aec43544cbb1ee4af39d352fe6595d627c77ce137e83c17895bdd56ee6a3986c2eef488a6027aceeb1ae5f155a6264d7d879d325091cc6bd59e8b5b47b560c1aedd5d20e542bb712cde98d34391b56a90b6aaec911dbf61a6c75d8333d4974941abcb86f29b840e31d4feb5bf92c0cd5799a2a805edb4c4b218889c7b863e14ebc1d856a625b557e5d1d8c6fd6c98076e4b1d0649686aab4a91c6879de588513b520d5f12427ce26d5a7a6f49e549f7ac5f331520d38f243a4fa406e47aaf775e3ebd8b4753a103fdd844e37f5c46adf660aef5e1d9e5b6dbed9f48e0b7fb369eceb86f6a7bb0aec9b4d3fdff0a0cb89f387f62d9e77ef3ebed9f46baf53b69754475c18c1194d8c675d731d69f866d3fff0faea28dedf6cfafc17b35fc3a64f9de110103e3bc89c6c0ab1e93fff02e014e2caab1e0000\"]"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Fri, 20 Sep 2019 17:36:01 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "status",
              "value": "200 OK"
            },
            {
              "name": "x-ratelimit-limit",
              "value": "5000"
            },
            {
              "name": "x-ratelimit-remaining",
              "value": "4877"
            },
            {
              "name": "x-ratelimit-reset",
              "value": "1569004552"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=60, s-maxage=60"
            },
            {
              "name": "vary",
              "value": "Accept, Authorization, Cookie, X-GitHub-OTP, Accept-Encoding"
            },
            {
              "name": "etag",
              "value": "W/\"0648775486f7737e26a8d07259fa331c\""
            },
            {
              "name": "last-modified",
              "value": "Fri, 13 Sep 2019 16:10:06 GMT"
            },
            {
              "name": "x-github-media-type",
              "value": "github.machine-man-preview; format=json"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "x-github-request-id",
              "value": "B08B:0E2A:DD8954:1810044:5D850E01"
            }
          ],
          "headersSize": 1138,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2019-09-20T17:36:01.133Z",
        "time": 591,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 591
        }
      },
      {
        "_id": "ccb37693035b0dfd294493b4c48b58eb",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/vnd.github.v3+json"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "jupiter-integration-github octokit-rest.js/17.6.0 octokit-core.js/2.5.4 Node.js/10.19.0 (Linux 4.4; x64)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "Bearer [REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.github.com"
            }
          ],
          "headersSize": 341,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.github.com/repos/github-app-test/power-demo/pulls/4/reviews"
        },
        "response": {
          "bodySize": 1193,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 1193,
            "text": "[\"1f8b08000000000000039d54db8eda3010fd97f495ac1313ae52557505ad7808d5964b1155153989134c9338b59db06cc4bfef38a115a0962d3c19323e67ce8ccfccf7ca60a131c44edb69f76d0bb78c8c87d4d3df0c77346e7f491e6dfff3f36cbdfa64af57536b912e31f9b66cbbdba797e9c875dcf9c7bdd1320a498531ac8c84c72c0368c4539a294630c43457c7eae29eddb7cfe99fbacbd53409b68bdd743b69bbf3f819ae93922822bc4224c0b3512a9743849a8fd27e8899da14be4e17f04c418e8780a7a840bf137c28df3b40128b234d5d8816784e97b3235303073a894e356f549a5c486832d7f74f6f463c49f80ef0978aafa6008a230cb435bf5916df4301b00a71b5a1d03228e3a08b67\",\"52dd28a78654481ff0f69a44c223081ade26e9080241bb0cb45448d09cd76c852f03c172c57876a3b43328507111938cbd903ba8002a81418bba51440d01282dc17237621b4c8572c14a12ec753b040d282ba1bbf7f05d80814eed730a365fe821845e33453d12a67a102392487a68193e0ff7cd20c013297d793499b993d96c3c02c475b737c637499e9b8a4a85726d7733a42947799124c879a70f417f151015b4647467fed927c0aea3de317cdd4e7597c1847ac4ff994f223de0a4501b2e3c22250f58ed05a8c91dbb8fe3af10f51296fd947a21e9d2ea53d0e8649f9c0cf31be9de280f7a7b5adf5f5291f345f0df351e801abc9f32a5c02844817a6cd903d3ea987830b7fb43dc1edaf61aaa859504b79a9d6d45833eb630eefafd81d3837fb847b01dfa7e1475a220c0837e14750776cf38fc78058f58b432fb050000\"]"
          },
          "cookies": [],
          "headers": [
            {
              "name": "server",
              "value": "GitHub.com"
            },
            {
              "name": "date",
              "value": "Fri, 20 Sep 2019 17:36:02 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "status",
              "value": "200 OK"
            },
            {
              "name": "x-ratelimit-limit",
              "value": "5000"
            },
            {
              "name": "x-ratelimit-remaining",
              "value": "4876"
            },
            {
              "name": "x-ratelimit-reset",
              "value": "1569004552"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=60, s-maxage=60"
            },
            {
              "name": "vary",
              "value": "Accept, Authorization, Cookie, X-GitHub-OTP"
            },
            {
              "name": "etag",
              "value": "W/\"902ee25e799b576893150dda096ec750\""
            },
            {
              "name": "x-github-media-type",
              "value": "github.machine-man-preview; format=json"
            },
            {
              "name": "access-control-expose-headers",
              "value": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubdomains; preload"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "origin-when-cross-origin, strict-origin-when-cross-origin"
            },
            {
              "name": "content-security-policy",
              "value": "default-src 'none'"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "x-github-request-id",
              "value": "F210:4B97:876380:141D236:5D850E01"
            }
          ],
          "headersSize": 1075,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2019-09-20T17:36:01.728Z",
        "time": 472,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 472
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
