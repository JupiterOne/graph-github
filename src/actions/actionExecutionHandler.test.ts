import actionExecutionHandler, {
  IntegrationActionExecutionContext,
} from './actionExecutionHandler';
import { IntegrationConfig, sanitizeConfig } from '../config';
import { integrationConfig } from '../../test/config';
import {
  createMockIntegrationLogger,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { setupGithubRecording } from '../../test/recording';

jest.setTimeout(30000);

let recording: Recording;
afterEach(async () => {
  await recording.stop();
});

describe('actionExecutionHandler', () => {
  test.skip('action.name=PARTIAL_INGEST', async () => {
    // Arrange
    recording = setupGithubRecording({
      directory: __dirname,
      name: 'partialIngest',
    });
    sanitizeConfig(integrationConfig);

    const executionContext = {
      event: {
        action: {
          name: 'PARTIAL_INGEST' as any,
          parameters: {
            entities: [
              {
                _key: 'j1-ingest/reimagined-barnacle/pull-requests/19',
                _type: 'github_pullrequest',
              },
              {
                _key: 'bob',
                _type: 'github_member',
              },
            ],
          },
        },
      },
      clients: {},
      instance: {
        config: integrationConfig,
      },
      logger: createMockIntegrationLogger(),
    } as unknown as IntegrationActionExecutionContext<IntegrationConfig>;

    // Act
    const result = await actionExecutionHandler(executionContext);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
