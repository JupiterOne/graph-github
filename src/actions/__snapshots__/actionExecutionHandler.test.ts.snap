// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionExecutionHandler action.name=PARTIAL_INGEST 1`] = `
Object {
  "entities": Array [],
  "errors": Array [
    Object {
      "_key": "bob",
      "_type": "github_member",
      "message": "github_pullrequest is the only supported entity type.",
    },
    Object {
      "_key": "j1-ingest/reimagined-barnacle/pull-requests/19",
      "_type": "github_pullrequest",
      "message": "Failed to collect data for entity.",
    },
    Object {
      "_key": "j1-ingest/reimagined-barnacle/pull-requests/190000",
      "_type": "github_pullrequest",
      "message": "Failed to collect data for entity.",
    },
  ],
}
`;
